python/ray/autoscaler/commands.py

```diff
@@ -84,33 +84,38 @@ def teardown_cluster(config_file, yes, workers_only, override_cluster_name):
     provider = get_node_provider(config["provider"], config["cluster_name"])
     try:
 
-        def remaining_nodes():
-            if workers_only:
-                A = []
-            else:
-                A = [
-                    node_id for node_id in provider.non_terminated_nodes({
-                        TAG_RAY_NODE_TYPE: NODE_TYPE_HEAD
-                    })
-                ]
-
-            A += [
+        def remaining_heads():
+            return [
+                node_id for node_id in provider.non_terminated_nodes({
+                    TAG_RAY_NODE_TYPE: NODE_TYPE_HEAD
+                })
+            ]
+
+        def remaining_workers():
+            return [
                 node_id for node_id in provider.non_terminated_nodes({
                     TAG_RAY_NODE_TYPE: NODE_TYPE_WORKER
                 })
             ]
-            return A
 
         # Loop here to check that both the head and worker nodes are actually
         #   really gone
-        A = remaining_nodes()
+        head = remaining_heads()
+        A = remaining_workers()
```

Comment:
How about we fix the underlying bug in AWS node provider instead? It can internally split up the stop request.

Reply:
I see; is it reasonable to assume head = 1 node? 

I wouldn't want to check for spot instance on every single node ID passed in because presumably that's slow? especially if not cached?

Another option here is to just take the top 5 and see if there's any spot instances in there, and if so, then terminate everybody - and otherwise to stop everybody?

Reply:
I think they are all cached, you can check them all very quickly.

Reply:
OK